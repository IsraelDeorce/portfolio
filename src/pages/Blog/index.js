export { default as BigO } from './BigO'
export { default as DataStructures } from './DataStructures'
export { default as Algorithms } from './Algorithms'
export { default as SoftwareEngineering } from './SoftwareEngineering'
