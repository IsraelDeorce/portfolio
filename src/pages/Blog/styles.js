import styled from 'styled-components'

export const Blog = styled.div`
  display: flex;
  flex-direction: column;
`
