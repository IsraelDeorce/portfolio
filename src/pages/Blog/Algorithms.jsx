import * as S from './styles'

function Algorithms() {

  return (
    <S.Blog>
      <h1>Algoritms</h1>
      <p>Under construction...</p>
    </S.Blog>
  )
}

export default Algorithms
