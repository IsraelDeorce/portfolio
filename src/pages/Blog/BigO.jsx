import * as S from './styles'

function BigO() {

  return (
    <S.Blog>
      <h1>Big O</h1>
      <p>Under construction...</p>
    </S.Blog>
  )
}

export default BigO
