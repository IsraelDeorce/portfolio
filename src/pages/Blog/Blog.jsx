import * as S from './styles'

function Blog() {

  return (
    <S.Blog>
    </S.Blog>
  )
}

export default Blog
