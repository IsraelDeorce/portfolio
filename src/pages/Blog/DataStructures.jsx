import * as S from './styles'

function DataStructures() {

  return (
    <S.Blog>
      <h1>Data Structures</h1>
      <p>Under construction...</p>
    </S.Blog>
  )
}

export default DataStructures
