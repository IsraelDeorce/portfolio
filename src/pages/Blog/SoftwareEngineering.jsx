import * as S from './styles'

function SoftwareEngineering() {

  return (
    <S.Blog>
      <h1>Software Engineering</h1>
      <S.Centralized>
        <p><cite>Humanity is now faced with a new set of challengesâ€”climate change and extreme
          weather, declining natural resources, an increasing world population to be fed and
          housed, international terrorism, and the need to help elderly people lead satisfying
          and fulfilled lives. We need new technologies to help us address these problems and,
          for sure, software will play a central role in these technologies.</cite></p>
        <p>&mdash; Ian Sommervile, 2010.</p>
      </S.Centralized>
      <h1>Process Management</h1>
      <p>Lorem ipsum</p>
      <h1>Analysis</h1>
      <h1>Architecture and Design</h1>
      <h1>Development</h1>
      <h1>Testing</h1>
      <h1>Maintenance</h1>
    </S.Blog>
  )
}

export default SoftwareEngineering
