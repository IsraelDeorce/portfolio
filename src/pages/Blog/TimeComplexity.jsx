import * as S from './styles'

function TimeComplexity() {

  return (
    <S.Blog>
      <h1>Time Complexity</h1>
      <p>Under construction...</p>
    </S.Blog>
  )
}

export default TimeComplexity
