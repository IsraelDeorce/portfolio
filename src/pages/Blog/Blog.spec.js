import { useHistory } from 'react-router-dom'
import Blog from './Blog'

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(() => ({}))
}))

describe('<Blog/>', () => {
  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<Blog />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<Blog />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<Blog />)).toMatchSnapshot()
    })
  })
})
