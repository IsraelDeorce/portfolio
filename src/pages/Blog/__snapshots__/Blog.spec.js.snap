// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog/> Snapshot renders without crashing using mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<Blog>
  <styled.div>
    <div
      className="c0"
    >
      <h1>
        TEST PAGE
      </h1>
    </div>
  </styled.div>
</Blog>
`;

exports[`<Blog/> Snapshot renders without crashing using render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
>
  <h1>
    TEST PAGE
  </h1>
</div>
`;

exports[`<Blog/> Snapshot renders without crashing using shallow 1`] = `
<styled.div>
  <h1>
    TEST PAGE
  </h1>
</styled.div>
`;
